shell/platform/android/io/flutter/embedding/engine/systemchannels/SettingsChannel.java

```diff
@@ -17,13 +21,27 @@ public class SettingsChannel {
   private static final String BRIEFLY_SHOW_PASSWORD = "brieflyShowPassword";
   private static final String ALWAYS_USE_24_HOUR_FORMAT = "alwaysUse24HourFormat";
   private static final String PLATFORM_BRIGHTNESS = "platformBrightness";
+  private static final String CONFIGURATION_GENERATION = "configurationId";
 
   @NonNull public final BasicMessageChannel<Object> channel;
 
   public SettingsChannel(@NonNull DartExecutor dartExecutor) {
     this.channel = new BasicMessageChannel<>(dartExecutor, CHANNEL_NAME, JSONMessageCodec.INSTANCE);
   }
 
+  public static DisplayMetrics getPastDisplayMetrics(int configGeneration) {
+    final LinkedList<SentConfiguration> pastConfigurations = SentConfiguration.sentConfigurations;
+    while (pastConfigurations.size() > 0
+        && pastConfigurations.getFirst().generationNumber != configGeneration) {
+      pastConfigurations.remove();
+    }
```

Comment:
Some of the higher level doc comments should mention that you cannot request font scaling for arbitrary generations, you have to use the latest generation you saw.

Reply:
FWIW, the naming `generation` to me seems a little more fitting, since this is not a classical ID.

Reply:
well I'll go with generation here and everywhere else then.